### PR TITLE
Use debuginfo-install for debuginfo packages

### PR DESCRIPTION
**Pull request by martinky82**

### CODE REVIEW

File: tmt/steps/prepare/install.py [hunk 1]
```diff
@@ -165,17 +165,25 @@ def go(self, guest):
         packages = self.get('package', [])
         repo_packages = []
         local_packages = []
+        repo_debuginfos = []
```
Comment (psss) [inline tmt/steps/prepare/install.py:168]: What about calling it `debuginfo_packages` to make the content more obvious?

File: tmt/steps/prepare/install.py [hunk 2]
```diff
@@ -165,17 +165,25 @@ def go(self, guest):
...
+        # and split debug{info,source} if installing from repo
         for package in packages:
             if package.endswith('.rpm'):
                 local_packages.append(package)
             else:
-                repo_packages.append(package)
+                if 'debuginfo' in package or 'debugsource' in package:
+                    # strip -debuginfo from package name
+                    # installing with it doesn't work on RHEL7
+                    package.replace('-debuginfo','')
```
Comment (psss) [inline tmt/steps/prepare/install.py:183]: Replace does not work in-place, it returns the modified string. So the original package name is stored instead.

File: tmt/steps/prepare/install.py [hunk 3]
```diff
@@ -165,17 +165,25 @@ def go(self, guest):
...
             return
 
         # Detect local packages and directories
+        # and split debug{info,source} if installing from repo
         for package in packages:
             if package.endswith('.rpm'):
                 local_packages.append(package)
             else:
-                repo_packages.append(package)
+                if 'debuginfo' in package or 'debugsource' in package:
```
Comment (psss) [inline tmt/steps/prepare/install.py:180]: Would be probably safer to use `re` or `str.endswith()` and match end of line here, quick check shows there might be packages with `debuginfo` in the middle of the name such as `elfutils-debuginfod-client`.

File: tmt/steps/prepare/install.py [hunk 4]
```diff
@@ -238,3 +246,21 @@ def go(self, guest):
...
+            # Show a brief summary by default
+            if not self.opt('verbose'):
+                summary = fmf.utils.listed(repo_debuginfos, max=3)
+                self.info('package', summary, 'green')
+            # Provide a full list of packages in verbose mode
+            else:
+                summary = fmf.utils.listed(repo_debuginfos, 'package')
+                self.info('package', summary + ' requested', 'green')
+                for package in sorted(repo_debuginfos):
+                    self.verbose(package, shift=1)
```
Comment (psss) [inline tmt/steps/prepare/install.py:261]: This summary part code is almost completely duplicated from above. Let's create a new private method `_package_summary` and move the shared code for both `repo_packages` and `debuginfo_packages` there.

File: tmt/steps/prepare/install.py [hunk 5]
```diff
@@ -238,3 +246,21 @@ def go(self, guest):
...
+            # Quote package names and prepare the rpm check
+            packages = ' '.join(
+                [tmt.utils.quote(package) for package in repo_debuginfos])
+            check = f'rpm -q --whatprovides {packages}'
+            guest.execute(f"{check} || debuginfo-install -y {repo_debuginfos}")
```
Comment (psss) [inline tmt/steps/prepare/install.py:266]: I think the `debuginfo-install` command does not always have to be installed on the box. We should probably include `{command} install -y /usr/bin/debuginfo-install` to ensure the command is available.

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for implementing this. Looks good in general. Added a couple of comments. Could you please also extend the `/tests/prepare/install` test with at least a simple coverage for this new feature? Thanks.

File: tmt/steps/prepare/install.py (hunk 3 above)
Comment (martinky82) [reply tmt/steps/prepare/install.py:180]: That exactly what I was thinking about. debugifno indeed isn't always in the end. My example was perl-debuginfo.i686 on x86_64 system.

File: tmt/steps/prepare/install.py (hunk 4 above)
Comment (martinky82) [reply tmt/steps/prepare/install.py:261]: Agreed, will do.

File: tmt/steps/prepare/install.py (hunk 1 above)
Comment (martinky82) [reply tmt/steps/prepare/install.py]: Done. Thanks

File: tmt/steps/prepare/install.py (hunk 2 above)
Comment (martinky82) [reply tmt/steps/prepare/install.py]: Fixed, thanks for the catch.

File: tmt/steps/prepare/install.py (hunk 3 above)
Comment (martinky82) [reply tmt/steps/prepare/install.py:197]: I'll leave it as is. Although str.endswith would be safe, this approach gives is additional bit of flexibility.

File: tmt/steps/prepare/install.py (hunk 4 above)
Comment (martinky82) [reply tmt/steps/prepare/install.py]: Done, thanks

File: tmt/steps/prepare/install.py (hunk 5 above)
Comment (martinky82) [reply tmt/steps/prepare/install.py]: Done, thanks.

File: tmt/steps/prepare/install.py [hunk 6]
```diff
@@ -238,3 +250,11 @@ def go(self, guest):
...
                 f"{check} || {command} install {options} "
                 f"{packages}{yum_check}")
+
+        # Install debug{info,source} from repos
+        if debuginfo_packages:
+            packages = __prepare_install(debuginfo_packages)
+
+            # make sure debuginfo-install is present on the target system
+            guest.execute(f"{command} install -y /usr/bin/debuginfo-install}")
+            guest.execute(f"{check} || debuginfo-install -y {packages}")
```
Comment (psss) [inline tmt/steps/prepare/install.py:260]: The `check` part cannot be used here as debuginfo packages have stripped names.

Comment (martinky82) [reply tmt/steps/prepare/install.py]: Indeed. Removed

**Review by psss — APPROVED** [review]
Thanks for the improvements. I've added a couple of adjustments, especially to hadle the special cases like `elfutils-debuginfod-debuginfo` which would have the name incorrectly modified. Test coverage extended accordingly. For CentOS 8 the repositories are not correctly enabled, filed [BZ#1964505](https://bugzilla.redhat.com/show_bug.cgi?id=1964505) to investigate this. Commits squashed, should be ready to merge if tests pass.

**Pull request merged**